src/Psalm/Internal/Provider/ReturnTypeProvider/ExceptionCodeReturnTypeProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Psalm\Internal\Provider\ReturnTypeProvider;
+
+use Psalm\Plugin\EventHandler\Event\MethodReturnTypeProviderEvent;
+use Psalm\Plugin\EventHandler\MethodReturnTypeProviderInterface;
+use Psalm\Type;
+use PDOException;
+
+use function is_a;
+
+class ExceptionCodeReturnTypeProvider implements MethodReturnTypeProviderInterface
+{
+    public static function getClassLikeNames(): array
+    {
+        return ['Throwable'];
+    }
+
+    public static function getMethodReturnType(MethodReturnTypeProviderEvent $event): ?Type\Union
+    {
+        $method_name_lowercase = $event->getMethodNameLowercase();
+        $fqcn = $event->getCalledFqClasslikeName();
+
+        if ($method_name_lowercase !== 'getcode') {
+            return null;
+        }
+
+        if ($fqcn === 'Exception' || $fqcn === 'Throwable') {
+            return null;
+        }
+
+        if (is_a($fqcn, PDOException::class, true)) {
```

Comment:
Please check with the string instead. I'm not sure there are childs of PDOException out there, and I'd prefer avoid triggering the autoloader when it's not needed

Reply:
What if someone extends the PDOException ?

Reply:
Let's worry about that if it happens (though I'd be surprised it does), I had bad surprises before when trying to call the autoloader to perform static analysis (it crashed Psalm when the class did not exist)

If you still want to do it, I think you should be able to check it statically, by fetching the ClassLikeStorage and checking its parents recursively

Reply:
See also #7107, I would like to eventually get to the point where Psalm can analyze most extensions without them being loaded at runtime.